packages/accounts/src/msca/plugins/types.ts

```diff
@@ -0,0 +1,6 @@
+import type { BaseSmartContractAccount } from "@alchemy/aa-core";
+
+export interface Plugin<D> {
+  meta: { name: string; version: string };
+  accountDecorators: (a: BaseSmartContractAccount) => D;
```

Comment:
we should expose a method or the `rpcProvider` property on `ISmartContractAccount` so we can use that here instead of `BaseSmartContractAccount`

Reply:
by having made it readonly you can now, right?

Reply:
I did but not on the ISmartContractAccount, I can update the ISmartContractAccount interface here if we're cool with that change